docs/0.5/tutorials/using_the_admin_ui.md

```diff
@@ -23,21 +23,91 @@ then propose a circuit and view the proposal summary. As Bob, you'll register
 and add keys, vote on the circuit proposal, then view the new circuit's summary.
 Finally, you'll log out for both Alice and Bob.
 
+There are two paths in this tutorial: the first uses Biome for user
+authentication, and the second uses OAuth. Biome authentication will require
+users to register and login with a username and password that is verified by
+splinterd, while OAuth authentication uses a configured OAuth provider for
+logging users in. Both of these options will be covered. The choice here only
+affects the initial configuration and user registration/login; the other steps
+of this tutorial are the same for both paths.
+
 ## Prerequisites
 
 * A clone of the [splinter-ui repository](https://github.com/Cargill/splinter-ui)
+
 * [Docker Engine](https://docs.docker.com/engine/)
+
 * [Docker Compose](https://docs.docker.com/compose/)
 
+* If using OAuth authentication, you will need to register an OAuth app with a
+  supported OAuth provider. Splinter supports GitHub and any OAuth provider that
+  conforms to the OpenID standard, including Google and Azure AD. Here are
+  instructions for registering an app with some of the supported providers:
+
+  * [Registering a GitHub OAuth app](https://docs.github.com/en/free-pro-team@latest/developers/apps/creating-an-oauth-app)
+
+  * [Registering a Google OAuth app](https://support.google.com/cloud/answer/6158849)
+
+  * [Registering an Azure AD OAuth app](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app)
+
+  Some providers will require a homepage URL and a redirect URL; if these are
+  required, use the values `http://localhost:8080` and
+  `http://localhost:8080/splinterd/oauth/callback`, respectively.
```

Comment:
Just ran through this with google openid oauth setup and in this example the google oauth app would require two redirect URLs: `http://localhost:3030/splinterd/oauth/callback` and `http://localhost:3031/splinterd/oauth/callback`

Reply:
Good catch. I removed the port numbers from these URLs. Most OAuth providers (including GitHub and Google) treat localhost as a special case where the port can be anything. So `http://localhost/splinterd/oauth/callback` works for both acme and bubba nodes.